paper/paper.md

```diff
@@ -0,0 +1,84 @@
+
+Instructors in many fields have already begun using the Jupyter notebook as a teaching platform. The notebook has appeared in over 70 classes [@Castano_Jupyter_Map_Dataset] on subjects including geology, mathematics, mechanical engineering, data science, chemical engineering, and bioinformatics, just to name a few. Software Carpentry, which aims to teach graduate students basic computational skills, has also adopted the notebook for some of its lessons [@Wilson2014].
+
+Despite its appearance in many classrooms—prior to the existence of nbgrader—the notebook was rarely used on a large scale for *graded* assignments. Instead, it was often used either for ungraded in-class exercises; or in classes small enough that notebooks can be graded by hand. This is because there are several challenges to using the notebook for graded assignments at scale. First, for large class sizes, it is not feasible for an instructor to manually grade the code that students write: there must be a way of autograding the assignments. A Jupyter notebook is not a typical script that can be run and may contain multiple parts of a problem within the same notebook. Second, for many courses, the programming is a means to an end for understanding concepts in a specific domain. Instructors may also want students to provide written free-responses interpreting the results of their code, and thus need to be able to rely on autograding for the coding parts of an assignment, but also be able to manually grade the written responses in the surrounding context of the student's code. Third, the process of distributing assignments to students and later collecting them can be tedious, and this is true even more so with Jupyter notebooks because there is a separate interface for accessing them beyond the standard system file browser. This often leads to confusion on the part of students about how to open notebooks after downloading them, and where to find the notebooks in order to submit them.
+
+nbgrader streamlines the repetitive tasks found in course management and grading, and its flexibility allows greater communication between instructor and student. Overall, nbgrader improves the learning experience as students and instructors can focus on content and building understanding by minimizing or automating the tedious and repetitive tasks.
```

Comment:
- [ ] `as students and instructors` => `because students and instructors`
- [ ] "by minimizing or automating the tedious and repetitive tasks." => "by minimizing or automating the tedious and repetitive tasks associated with grading."

Reply:
- [ ] `nbgrader improves the learning experience` => `nbgrader improves the learning experience for both instructors and students` (this foreshadows to the two examples you include in the second part of the sentence)